pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -752,7 +752,18 @@ func getTaskrunWorkspaces(pr *v1beta1.PipelineRun, rprt *resources.ResolvedPipel
 			}
 			workspaces = append(workspaces, taskWorkspaceByWorkspaceVolumeSource(b, taskWorkspaceName, pipelineTaskSubPath, pr.GetOwnerReference()))
 		} else {
-			return nil, "", fmt.Errorf("expected workspace %q to be provided by pipelinerun for pipeline task %q", pipelineWorkspaceName, rprt.PipelineTask.Name)
+			workspaceIsOptional := false
+			if rprt.ResolvedTaskResources != nil && rprt.ResolvedTaskResources.TaskSpec != nil {
```

Comment:
@sbwsg this check covers the optional `workspaces` at the task level, do we need one more check for the optional workspaces at the pipeline level or are they resolved and covered by this check?

Reply:
We don't need to validate the pipeline workspaces in the same way.  It's valid for a task workspace to be optional when the pipeline workspace is not.  But it's not valid for a pipeline workspace to be optional when the task workspace is not (which is what we check here).